File: solutions/cpp/allergies/1/allergies.cpp
#include "allergies.h"

namespace allergies {
    std::vector<std::string> items{
        "eggs",
        "peanuts",
        "shellfish",
        "strawberries",
        "tomatoes",
        "chocolate",
        "pollen",
        "cats"
    };
    
    allergy_test::allergy_test(int test){
        for(int n = 0; n < 8; ++n){
            if(test&(1<<n)){
                this->results.insert(items[n]);
            }
        }
    }

    bool allergy_test::is_allergic_to(const std::string& test){
        for(const std::string& item : this->results){
            if(test == item){
                return true;
            }
        }
        return false;
    }
}  // namespace allergies

File: solutions/csharp/allergies/1/Allergies.cs
[Flags]
public enum Allergen : byte
{
    Eggs = 0b0000_0001,
    Peanuts = 0b0000_0010,
    Shellfish = 0b0000_0100,
    Strawberries = 0b0000_1000,
    Tomatoes = 0b0001_0000,
    Chocolate = 0b0010_0000,
    Pollen = 0b0100_0000,
    Cats = 0b1000_0000,
}

public class Allergies
{
    private readonly int mask;

    public Allergies(int mask)
    {
        this.mask = mask;
    }

    public bool IsAllergicTo(Allergen allergen)
        => (mask & (int)allergen) == (int)allergen;

    public Allergen[] List()
        => Enum.GetValues<Allergen>()
            .Where(allergen => IsAllergicTo(allergen))
            .ToArray();
    
}